camel-chimera-mapping-template/src/main/java/com/cefriel/rdf/MaptTemplateProcessor.java

```diff
@@ -93,92 +93,140 @@ private static boolean validateParams(OperationParams params) {
 
             // if filename is specified then save to file
             if(params.outputFileName() != null) {
                 new File(params.basePath()).mkdirs();
                 Path outputFilePath = Paths.get(params.basePath() + params.outputFileName());
                 if(params.query() != null) {
-                    List<Path> resultFilesPaths =
-                            templateExecutor.executeMappingParametric(reader, ResourceAccessor.open(params.template(), exchange),
-                                    ResourceAccessor.open(params.query(), exchange), outputFilePath, templateMap, formatter, usedTemplateFunctions);
+                    List<Path> resultFilesPaths;
+                    if (reader != null) {
+                        resultFilesPaths = templateExecutor.executeMappingParametric(Map.of("reader", reader),
+                                ResourceAccessor.open(params.template(), exchange),
+                                ResourceAccessor.open(params.query(), exchange),
+                                outputFilePath);
+                    }
+                    else {
+                        resultFilesPaths = templateExecutor.executeMappingParametric(readers,
+                                ResourceAccessor.open(params.template(), exchange),
+                                ResourceAccessor.open(params.query(), exchange),
+                                outputFilePath);
+                    }
                     exchange.getMessage().setBody(resultFilesPaths, List.class);
+
                 } else {
-                    Path resultFilePath =
-                            templateExecutor.executeMapping(reader, ResourceAccessor.open(params.template(), exchange), outputFilePath, templateMap, formatter, usedTemplateFunctions);
+                    Path resultFilePath;
+                    if (reader != null) {
+                        resultFilePath = templateExecutor.executeMapping(Map.of("reader", reader), ResourceAccessor.open(params.template(), exchange), outputFilePath);
+                    }
+                    else {
+                        resultFilePath = templateExecutor.executeMapping(readers, ResourceAccessor.open(params.template(), exchange), outputFilePath);
+                    }
+
                     exchange.getMessage().setBody(resultFilePath, String.class);
                 }
             }
             // if filename is not specified then the result of applying template is stored in the exchange body as string
             else {
                 if (params.query() != null) {
-                    Map<String,String> result = templateExecutor.executeMappingParametric(reader, ResourceAccessor.open(params.template(), exchange), ResourceAccessor.open(params.query(), exchange), templateMap, formatter, usedTemplateFunctions);
+                    Map<String,String> result;
+                    if (reader != null) {
```

Comment:
Here and in other invocations to the `mapping-template`, avoid repeated code blocks. Change as:

```java
Map<String, Reader> readers; //Check where defined and if we need to define it again
if (reader != null)
    readers = Map.of("reader", reader);
result = templateExecutor.executeMappingParametric(readers,
                                ResourceAccessor.open(params.template(), exchange),
                                ResourceAccessor.open(params.query(), exchange));
```